integration-tests/test.sh

```diff
@@ -46,7 +47,7 @@ echo "$out"
 if [[ $os == "arch" ]]; then
   egrep -q 'Count: 74, Failed: 0' <<<"$out"
 else
-  egrep -q 'Count: 88, Failed: 0' <<<"$out"
+  egrep -q 'Count: 71, Failed: 0' <<<"$out"
```

Comment:
In upstream CI, 88 tests are running and passing according to logs there, for this conditional. Since the upstream is also running in travis, that's ... weird.

https://api.travis-ci.org/v3/job/473850963/log.txt

Reply:
Yep, I've removed some of the dnstest.io tests, that's why it's less here now.

Reply:
Yes - my question is why is this passing in upstream, but not here, when presumbably there's nothing different about the travis environment the build is running inside.

Reply:
I don't understand at all. It works from in-office?

```
$ dig mx dnstest.io

; <<>> DiG 9.10.6 <<>> mx dnstest.io
;; global options: +cmd
;; Got answer:
;; ->>HEADER<<- opcode: QUERY, status: NOERROR, id: 19388
;; flags: qr rd ra; QUERY: 1, ANSWER: 2, AUTHORITY: 0, ADDITIONAL: 1

;; OPT PSEUDOSECTION:
; EDNS: version: 0, flags:; udp: 4096
;; QUESTION SECTION:
;dnstest.io.            IN    MX

;; ANSWER SECTION:
dnstest.io.        3599    IN    MX    10 b.dnstest.io.
dnstest.io.        3599    IN    MX    5 a.dnstest.io.

;; Query time: 37 msec
;; SERVER: 172.16.116.1#53(172.16.116.1)
;; WHEN: Thu Apr 04 17:44:43 BST 2019
;; MSG SIZE  rcvd: 75
```

Reply:
It's probably cached or MX is not handled by authoritative nameservers for dnstest.io. Try running `dig +trace -t SRV _https._tcp.dnstest.io` a couple of times and you'll see different results.

Reply:
```
$ dig +trace -t SRV _https._tcp.dnstest.io

; <<>> DiG 9.10.6 <<>> +trace -t SRV _https._tcp.dnstest.io
;; global options: +cmd
.			427033	IN	NS	b.root-servers.net.
.			427033	IN	NS	h.root-servers.net.
.			427033	IN	NS	i.root-servers.net.
.			427033	IN	NS	e.root-servers.net.
.			427033	IN	NS	f.root-servers.net.
.			427033	IN	NS	k.root-servers.net.
.			427033	IN	NS	a.root-servers.net.
.			427033	IN	NS	l.root-servers.net.
.			427033	IN	NS	g.root-servers.net.
.			427033	IN	NS	m.root-servers.net.
.			427033	IN	NS	c.root-servers.net.
.			427033	IN	NS	j.root-servers.net.
.			427033	IN	NS	d.root-servers.net.
.			517703	IN	RRSIG	NS 8 0 518400 20190418050000 20190405040000 25266 . VNWzDcvX06igl5CFfUiI17m1zagzlUYd2+RSepxu/XU+4KXcPgCe0ZdT kUqePg9vudkrlyVutXe4kpv8WRuP30EKVDEt8kU0V+TxAfjUjYiR5lmd u8FuB+eIkbT5yT5NcS4xuY1W5nRdRTvTIgHUqWC2NZ2IrCVwlTcHLnra qbegu9rWlxrYh5kc6FS9/WLdCAHFxH+LIqqaPl7hHFA4PwJ0AgzYf4v2 Fz+SUX6te4AYdj/D3pBPnccoKYQp5gOinNHrYkxfDy3R6hVSpKq7d1bG ERheOLVqkD8vG9dAb21wu1vha+SBkVY9nRVm9A2ujdMEPA90zAUs0G4p 6RzW0A==
;; Received 525 bytes from 192.168.178.1#53(192.168.178.1) in 46 ms

io.			172800	IN	NS	c0.nic.io.
io.			172800	IN	NS	ns-a1.io.
io.			172800	IN	NS	b0.nic.io.
io.			172800	IN	NS	a2.nic.io.
io.			172800	IN	NS	a0.nic.io.
io.			172800	IN	NS	ns-a3.io.
io.			86400	IN	DS	57355 8 1 434E91E206134F5B3B0AC603B26F5E029346ABC9
io.			86400	IN	DS	57355 8 2 95A57C3BAB7849DBCDDF7C72ADA71A88146B141110318CA5BE672057 E865C3E2
io.			86400	IN	DS	64744 8 2 2E7D661097A76EAC145858E4FF8F3DDAE5EAEDFD527725BC6F8A943E 4FE23A29
io.			86400	IN	RRSIG	DS 8 1 86400 20190418050000 20190405040000 25266 . RHHCrov5qTj8aK9NpWZWVHfJtvMxGTDOuYcCSxFzGNiWG5mRKGU+oyTB oYRdWpEdwB9bhDYI5ZlEdk5ZP3PfZ3xjXn5hnkh9jqWJZO8pgthxc3mJ s3lQdwduWfX3aioFl4RvgtuVcHJQMY4j025DCDGlApoNaHT1Zblsq7n0 bh57r9KnG11Mhvxk1Ctm66bI/33EkTS2GVEswkpid5WUHQ8cwU7UUU5i NIBGQsxM565h7Ql1EigbAeKUXnpynRb/cat7TZOe42znT4n238beZqJE 1+gQ/v2qvM6awIEEzlSaqEIhl7N7m5FDpiF1Q/joF6PadNhX5RkMZKrl 5wFzpQ==
;; Received 818 bytes from 202.12.27.33#53(m.root-servers.net) in 144 ms

dnstest.io.		86400	IN	NS	ns1.dnstest.io.
dnstest.io.		86400	IN	NS	ns2.dnstest.io.
dnstest.io.		86400	IN	DS	41961 7 1 DA0B7004175CDD360AFB10DA79B20BA67C463846
dnstest.io.		86400	IN	RRSIG	DS 8 2 86400 20190422182808 20190401172808 11766 io. IQxtKh5j4l7E3m1TF/4a9FaBpl3kfPQky+S6fsA7d5dmV8eEO5JaTrCm z004MN8htir1uUB4FZhgY9P1OirKyHMuHlNBra22USYIoXc4N54fiupS QlKCJrTWtXuUq/MkkQ3fCD/h0SrGjRVm2288nQC6nHvp1KSQmMIJ7UDD Qvs=
;; Received 337 bytes from 194.0.1.1#53(ns-a1.io) in 11 ms

_https._tcp.dnstest.io.	3600	IN	SRV	10 10 443 b.dnstest.io.
_https._tcp.dnstest.io.	3600	IN	SRV	0 5 443 a.dnstest.io.
_https._tcp.dnstest.io.	3600	IN	RRSIG	SRV 7 4 3600 20190315095038 20190215095038 34325 dnstest.io. X/hC59xQwabEKyIeUUzZA/y2F+MVIIWMSKj0p7MPAo5kcSKpeeARXddz r+up6v0U6/7fNQuhPcVnx4pEjgiHWtghiCnzHA3APp9rrXxgAIngp2SS zdaqO5PrRNdJ/NFKScSd6H8VcV4T6/lyYYa26D/D7vgSOCQYE+wszZ6N uhk=
dnstest.io.		3600	IN	NS	ns1.dnstest.io.
dnstest.io.		3600	IN	NS	ns2.dnstest.io.
dnstest.io.		3600	IN	RRSIG	NS 7 2 3600 20190315095038 20190215095038 34325 dnstest.io. If+nSUx01XyeA+cKbcodA5OOYWsHYXDgDhizPfvqHQ1uWS23e7BPL5Gy Ieg4FlSqfO2OfDBUA2+Dswi1rcCJTeLQasZ4CKevpFPF/VVPsQkUXwuK GcQJfdc5zreiWc2nknSKyWruHM/V71PmEV/q/knXab6wGt8JgvUz6/5n wTs=
;; Received 1239 bytes from 198.211.118.55#53(ns2.dnstest.io) in 17 ms
```

Also looks like success? This is from my home connection, no VPN etc.

Reply:
@petemounce when I'm running that dig trace, I sometimes get `couldn't get address for 'ns2.dnstest.io': failure`. This will then cause `host -t SRV _https._tcp.dnstest.io` to return SERVFAIL (that gets cached) and sometimes correct records.